src/test/java/com/spring/onlinebookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,271 @@
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto expected = bookDto;
+
+        BookDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(),
+                BookDto.class
+        );
+
+        EqualsBuilder.reflectionEquals(expected, actual, "id", "title",
+                "author", "price", "isbn", "categoryIds");
+    }
+
+    @Test
+    @DisplayName("Delete book by id")
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    void deleteBookById_BookExists_ReturnsNoContentStatus() throws Exception {
+        mockMvc.perform(
+                        delete(BOOK_URL + "/{id}", CORRECT_BOOK_ID)
+                                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNoContent())
+                .andReturn();
+    }
+
+    @Test
+    @DisplayName("Update book by id")
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    void updateBookById_BookExists_ReturnsBookDto() throws Exception {
+        String requestJson = objectMapper.writeValueAsString(updateBookRequestDto);
+        MvcResult result = mockMvc.perform(
+                        put(BOOK_URL + "/{id}", CORRECT_BOOK_ID)
+                                .content(requestJson)
+                                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto expected = new BookDto();
+        setUpdateBookRequestDtoToDto(updateBookRequestDto, expected);
+
+        BookDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(),
+                BookDto.class
+        );
+        EqualsBuilder.reflectionEquals(expected, actual, "id", "title",
+                "author", "price", "isbn", "categoryIds");
+    }
+
+    @Test
+    @DisplayName("Search books by params")
+    @WithMockUser(username = "user", roles = "USER")
+    void search_BookExists_ReturnsListBookDto() throws Exception {
+        SearchBookRequestDto searchParams = new SearchBookRequestDto(
+                new String[]{"title"},
+                null,
+                null
+        );
+
+        String requestJson = objectMapper.writeValueAsString(searchParams);
+        MvcResult result = mockMvc.perform(
+                        get(BOOK_URL)
+                                .content(requestJson)
+                                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        List<BookDto> expected = List.of(bookDto);
+
+        List<BookDto> actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(),
+                new TypeReference<>() {}
+        );
+
+        EqualsBuilder.reflectionEquals(expected, actual, "id", "title",
+                "author", "price", "isbn", "categoryIds");
+    }
+
+    private void initDto() {
```

Comment:
create TestUtil class and generate all test data there by using static methods
for all tests